sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -752,6 +774,14 @@ object RmmRapidsRetryIterator extends Logging {
  * `CpuSplitAndRetryOOM`, a split policy like `splitTargetSizeInHalfGpu` or
  * `splitTargetSizeInHalfCpu` can be used to retry the block with a smaller target size.
  */
-case class AutoCloseableTargetSize(targetSize: Long, minSize: Long) extends AutoCloseable {
+case class AutoCloseableTargetSize(targetSize: Long, minSize: Long) extends AutoCloseable
+  with RetrySizeAwareable {
+
+  override def sizeInBytes: Long = targetSize
+
   override def close(): Unit = ()
 }
+
+trait RetrySizeAwareable {
```

Comment:
Can we have some docs to explain what this is for and how it is intended to be used?

Reply:
Updated